src/lib/cps-budget-manager.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Krzysztof Frydryk <frydryk.krzysztof@intel.com>
+
+#include <rtos/spinlock.h>
+#include <sof/sof.h>
+#include <stdint.h>
+#include <sof/lib/cps-budget-manager.h>
+#include <rtos/clk.h>
+#include <sof/sof.h>
+#include <sof/math/numbers.h>
+#ifdef __ZEPHYR__
+#include <zephyr/sys/util.h>
+#endif
+
+static struct cps_budget_data cps_data;
```

Comment:
Given my previous statement, its worth saying in a comment that this should be in uncache (which it should be on Zephyr) so no one is tempted to move it to cache.

Reply:
is a comment there sufficient or do we have sth else (sparse?) for uncache specifically?

Reply:
@lgirdwood don't think so. An address space has to be specified explicitly for sparse to check it, all objects, that don't specify one explicitly are assumed to belong to the default address space - regardless whether static or automatic.